contrib/win32/win32compat/win32_dirent.c

```diff
@@ -99,12 +99,12 @@ struct dirent *readdir(void *avp)
 			return NULL;
 		}
 
-		strncpy(pdirentry->d_name, tmp, strlen(tmp) + 1);
+		strncpy(pdirentry->d_name, tmp, sizeof(pdirentry->d_name));
```

Comment:
why will the tmp size is greater than d_name?
the file names are limited by MAX_PATH (260 characters)?